sros2/test/sros2/test_cli.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2019 Apex.AI, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import tempfile
+
+from lxml import etree
+import pytest
+
+
+def test_cli_without_network():
+    # Test that the CLI tool still works even when libxml2 can't look up external resources
+    # because its been set to use a proxy that won't resolve.
+    # This test regressions of https://github.com/ros2/sros2/issues/157
+
+    env = os.environ.copy()
+    env['HTTP_PROXY'] = 'http://example.com'
+    tmp_dir = tempfile.mkdtemp(prefix='sros2_test')
+
+    subprocess.run(
```

Comment:
Should the api be used directly, to ensure we're test the expected library, rather whatever subprocess finds on the path? I think the other system tests using this pattern:
 https://github.com/ros2/sros2/blob/3689472750ecea5bccfd2cfdcf1e4a52d76f6ad0/sros2/test/sros2/commands/security/verbs/test_create_key.py#L31

Reply:
@ruffsl I'm sorry, but I can't answer this question for you. - I added this test based on your statement
```It might be handy to have a test that can assert the entire CLI works as expected when offline```

I can try to find a way to make sure that `subprocess` finds the right executable, but you're on both sides of the question "Should we invoke the API, or should we use the CLI"

Reply:
I think you should be calling the api, as this unit test is to check if the XML expansion works offline, not that the command line verb doesn't return a non-zero for some other reason.

Reply:
@ruffsl If your goal is to reduce the scope of the test to check that XML expansion works offline, would you accept running just `test_policy_to_permissions` with the HTTP_PROXY environment set?